runtime/src/main/java/org/corfudb/protocols/logprotocol/CheckpointEntry.java

```diff
@@ -162,6 +162,12 @@ void deserializeBuffer(ByteBuf b, CorfuRuntime rt) {
     @Override
     public void serialize(ByteBuf b) {
         super.serialize(b);
+
+        if (cpType == CheckpointEntryType.END
+                && getDict().get(CheckpointDictKey.START_LOG_ADDRESS) == null) {
+            throw new IllegalArgumentException(dict.get(CheckpointDictKey.START_LOG_ADDRESS));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "dict"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)